app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -248,10 +249,13 @@ def networks
 
   def sshkeys
     collector.sshkeys.each do |tkey|
+      cert = OpenSSL::X509::Certificate.new(tkey.ssh_key + "\n")
+      fingerprint = OpenSSL::Digest::SHA1.hexdigest(cert.to_der).scan(/../).join(':')
```

Comment:
`OpenSSL::X509::Certificate` is for an SSL cert where you have an ssh key, recommend

```suggestion
      fingerprint = OpenSSL::Digest::SHA1.hexdigest(tkey.ssh_key)
```

Reply:
Yeah but you don't want an X509 certificate so getting that line to not fail isn't very helpful :smile: , it looks like you followed https://stelfox.net/blog/2014/calculating-rsa-key-fingerprints-in-ruby/

> X509 certificates also use a key fingerprint to help identify a certificate's signing authority.

This isn't what you want to do here.  The closer analog to what you want to do is further down, `The next one I got working was the SSH fingerprints thanks to the RFCs mentioned earlier.`

But it is even easier than that, because you already get a formatted public key so you can completely skip the first two:
```
key = OpenSSL::PKey::RSA.new(File.read(path_to_key))
data_string = [7].pack('N') + 'ssh-rsa' + key.public_key.e.to_s(0) + key.public_key.n.to_s(0)
```
^ you get essentially these two "for free' because the API is returning this format to you already

Did you happen try my suggestion?

Reply:
I think the problem is that only the DER portion of the key is how the fingerprint is calculated, so the to_der is the important part that has to be kept.

So I _think_ this should work:

```suggestion
      ssh_key_der = OpenSSL::PKey::RSA.new(tkey.ssh_key).to_der
      fingerprint = OpenSSL::Digest::SHA1.hexdigest(ssh_key_der).scan(/../).join(':')
```

Reply:
Unless maybe I'm misunderstand what the structure of the `tkey.ssh_key` is?

Reply:
`tkey.ssh_key` is already a standard exported public key string so `OpenSSL::PKey::RSA.new(tkey.ssh_key)` isn't going to work.  Also note that would fail for non RSA keys.

Reply:
Ah ok, I was misunderstanding how the key was being returned by the API and assumed it was in PEM format.  Though I am now more confused about the DER format and the content of a .pub file.  I thought hexdigest needed the binary format.

Reply:
We could use `sshkey` gem https://github.com/bensie/sshkey#fingerprints

Reply:
Wow so I completely forgot that I did basically exactly this for GCE back in '15 https://github.com/ManageIQ/manageiq/pull/6109/files#diff-9f8ff9a0bafc734718f6453ef8e5994673ff993d54bb721e5e18ed13a8921b6aR307

Reply:
Oh then we should definitely use this gem and not roll our own.  @Kuldip-Nanda can you look into doing this the same way that @agrare did?

Reply:
So is this change as simple as this:

```suggestion
      fingerprint = SSHKey.sha1_fingerprint(tkey.ssh_key)
```

(as well as changing the require to `require 'ssh_key'`, and also adding ssh key to the gemspec?)